### PR TITLE
Home Page

### LINKED ISSUE #2: Video Listing

### PR DESCRIPTION
**Pull request by znjs**
- Added Nav Bar
- Added Video Listing
- Added Routing
- Added Aside and its navigation
- Added Loader State
explore

`Note: Preview link failed due to dependency issues, Kindly ignore`

### CODE REVIEW

File: src/components/Aside/Aside.jsx [hunk 1]
```diff
@@ -1,40 +1,51 @@
...
+        onClick={() => setDisplayLoader(true)}
+        style={({ isActive }) => (isActive ? bgClr : null)}
+        className="fx fx-ai-center f-1025 pd-b-05 cr-pt select-none clr-gray-50 brd-sm">
+        <i className="fa-solid fa-heart pd-i-1"></i>
+        <p>Liked Video</p>
+      </NavLink>
       <NavLink
         to="/history"
+        onClick={() => setDisplayLoader(true)}
         style={({ isActive }) => (isActive ? bgClr : null)}
```
Comment (ScoobyDubeyDo) [inline src/components/Aside/Aside.jsx:47]: **suggestion**: you could have used classNames in place of inline styles.

File: src/components/Explore/explore.css [hunk 2]
```diff
@@ -1,6 +1,4 @@
 .explore {
     height: calc(100vh - 56px);
```
Comment (simran61) [inline src/components/Explore/explore.css:2]: you can use Relative units instead of px

File: src/components/Explore/Explore.jsx [hunk 3]
```diff
@@ -1,27 +1,59 @@
...
 function Explore() {
+  const { videoState, setDisplayLoader } = useVideo();
+  const [playlistModalData, setPlaylistModalData] = useState({
+    showModal: false,
+    modalData: {},
+    video: {},
+  });
+  const [watchlaterVideos, setWatchlaterVideos] = useState([]);
+  const [fetchWatchlaterVideos, setFetchWatchlaterVideos] = useState(false);
+  const encodedToken = JSON.parse(localStorage.getItem("encodedToken"));
```
Comment (ScoobyDubeyDo) [inline src/components/Explore/Explore.jsx:18]: **suggestion**: here, in this case, `JSON.parse()` is not required as the token is already of type string

File: src/components/Explore/explore.css (hunk 2 above)
Comment (znjs) [reply src/components/Explore/explore.css:2]: Thank you for pointing it out will change it globally at the end

**Review by simran61 — APPROVED** [review]
- Nice UI
- Clean folder structure
- Easily readable code

File: src/components/ExploreCard/ExploreCard.jsx [hunk 4]
```diff
@@ -1,23 +1,76 @@
...
+
   return (
-    <div className="w-20 brd-sm o-hide clr-gray-50 bg-clr-gray-900 mg-05">
+    <div className="w-20 brd-sm o-hide clr-gray-50 bg-clr-gray-900 clr-gray-50 mg-05">
       <div className="p-rel">
         <img
           className="img-res cr-pt"
-          src="https://i.ytimg.com/vi/LRP8d7hhpoQ/hq720.jpg?sqp=-oaymwEcCNAFEJQDSFXyq4qpAw4IARUAAIhCGAFwAcABBg==&rs=AOn4CLAujWabke2prCLCyielJz3koNHJ0A"
+          src={`https://i.ytimg.com/vi/${video._id}/hq720.jpg`}
           alt=""
```
Comment (ScoobyDubeyDo) [inline src/components/ExploreCard/ExploreCard.jsx:27]: **must**: Alt attribute should not be empty

File: src/components/Navbar/navbar.css [hunk 5]
```diff
@@ -1,4 +1,17 @@
 .nav-options:hover {
     background-color: #4b5563;
     transition: 0.2s ease-in;
+}
+
+.global-search::placeholder {
```
Comment (ScoobyDubeyDo) [inline src/components/Navbar/navbar.css:6]: **suggestion**: placeholders are not user-selectable so this will not affect them in any way.

**Review by ScoobyDubeyDo — APPROVED** [review]
- the code is easy to understand
- UI also looks good
- there are a few suggestions that can be taken care of

File: src/components/Explore/Explore.jsx (hunk 3 above)
Comment (znjs) [reply src/components/Explore/Explore.jsx:18]: Thank you for pointing it out will change it globally at the end

File: src/components/ExploreCard/ExploreCard.jsx (hunk 4 above)
Comment (znjs) [reply src/components/ExploreCard/ExploreCard.jsx:27]: Will update it

File: src/components/Navbar/navbar.css (hunk 5 above)
Comment (znjs) [reply src/components/Navbar/navbar.css:6]: they are actually if you drag your cursor from outside the element

**Pull request merged**